### PR TITLE
video overlay window

### PR DESCRIPTION
**Pull request by gsambrotta**
User will be able to see video/call in a new window, clicking on a button.
Once video/call is in a different window, main video/call page display dark screen with feedbacks to remind user that the call is in a separate window.

Separate window will have the minimal design:
- person who is speaking's screen/video
- mute and video command
- close window button

Issues #230

### CODE REVIEW

File: js/app.js [hunk 1]
```diff
@@ -189,6 +189,40 @@
...
+				var width = 405;
+				var top = (window.screen.height - height) - 5 ;
+				var left = (window.screen.width - width) - 5;
+				var popStile = 'top='+top+', left='+left+', width='+width+', height='+height+', location=no, status=no, menubar=no, toolbar=no, titlebar=0, scrollbars=no, resizable=no';
+				var roomUrl = window.location.href;
+				var newWindow = window.open(roomUrl, '', popStile);
+
+				setTimeout( function(){
+					populateNewWindow(newWindow);
+				}, 4000);
```
Comment (ChristophWurst) [inline js/app.js]: why do we need the delay and why is it `4000`? Is that a magic number? 😁

File: js/app.js [hunk 2]
```diff
@@ -189,6 +189,40 @@
...
+				newWindow.document.body.append(app);
+				newWindow.document.body.setAttribute('class', 'separate-window');
+
+				$(btnPop).addClass('hidden');
+				$(btnClose).removeClass('hidden');
+				newWindow.addEventListener("blur", window.focus());
+			}
+
+			$('#close-separateWindow').on( 'click', function(evt) {
+				window.close();
```
Comment (ChristophWurst) [inline js/app.js]: Does that always work? I remember using it once but it wasn't really reliable across different browsers. Some browsers simply ignored it.

File: js/app.js [hunk 3]
```diff
@@ -189,6 +189,40 @@
...
+				newWindow.document.body.innerHTML = '';
+				newWindow.document.body.append(app);
+				newWindow.document.body.setAttribute('class', 'separate-window');
+
+				$(btnPop).addClass('hidden');
+				$(btnClose).removeClass('hidden');
+				newWindow.addEventListener("blur", window.focus());
+			}
+
+			$('#close-separateWindow').on( 'click', function(evt) {
```
Comment (ChristophWurst) [inline js/app.js]: `evt` param can be removed if not used 😉

File: templates/index.php [hunk 4]
```diff
@@ -37,6 +37,8 @@
 	<div id="app-content" class="participants-1">
 
 		<button id="video-fullscreen" class="icon-fullscreen-white" data-placement="bottom" data-toggle="tooltip" data-original-title="<?php p($l->t('Fullscreen')) ?>"></button>
+		<button id="video-separateWindow"> POPUP </button>
```
Comment (ChristophWurst) [inline templates/index.php]: missing l10n: `<?php p($l->t('Popup')) ?>`

File: templates/index.php [hunk 5]
```diff
@@ -37,6 +37,8 @@
...
+		<button id="close-separateWindow" class="hidden"> Close </button>
```
Comment (ChristophWurst) [inline templates/index.php]: missing l10n: `<?php p($l->t('Close')) ?>`

**Comment by gsambrotta** [comment]
@ChristophWurst thanks for all your feedbacks. Is still very WIP so most of your feedbacks will be fixed later. 
I will push soon something new :)

**Comment by ChristophWurst** [comment]
> @ChristophWurst thanks for all your feedbacks. Is still very WIP so most of your feedbacks will be fixed later.
> I will push soon something new :)

Oh right, didn't see the `2. developing` label 🙈 Nice work!

**Comment by jancborchardt** [comment]
@gsambrotta @Ivansss @nickvergessen @ChristophWurst what’s the state of this regarding Nextcloud 12? :) (Just as info: We have a feature freeze on next Wednesday and a first beta likely on Thursday.)

**Comment by nickvergessen** [comment]
Closing for now to allow better open task management.
Feel free to reopen, when you want to pick up the work again.

**Pull request closed without merging**